### DISCUSSION TITLE
Camera.stream

### DISCUSSION THREAD

**Discussion by adandl** [briis/unifiprotect#190] [Q&A]
Hi!
Has anyone managed to set up UniFi protect to support camera.stream service?

when I try I get this error message:

`camera.front does not support play stream service`

**Comment by briis** [comment]
Do you have stream: added to configuration.yaml?

**Reply by adandl** [reply]
Yes. No options though.

`stream:`

**Comment by briis** [comment]
So please describe your setup, and exactly what is not working. What service are you trying to call that does not work? Or is it something else?

**Reply by adandl** [reply]
Hello Briis!
Here is what I want to do.
run the service to play the live stream to my Apple TV
`camera.play_stream
entity_id: camera.front
media_player: media_player.wohnzimmer_2
`
According to this link:
https://www.home-assistant.io/integrations/camera/

It says the stream-service has to be configured and the camera must support.

Can you try if this works in your setup?

**Comment by briis** [comment]
I just tested this on a Apple TV media Player. And it works fine with the G4 series Cameras, but not with the G3 Series. (This is the same behaviour we seen with Safari browsers, so is most likely an Apple issue.
But the stream itself works fine with most of my cameras. What type of device are you trying to stream to as media_player?

**Reply by adandl** [reply]
Hi. I tried with a G3. 
finally a reason to upgrade to G4:)

**Reply by sjftech** [reply]
Hi, is this a limitation for G3 Series Cameras regardless of the player? I'm trying to use this on a LG WebOS TV and getting the same as adandl above so it doesn't seem like it's restricted to just Apple Media Players? The camera's show up fine in the overview tab however.

**Reply by AngellusMortis** [reply]
As long as streaming works inside of the WebUI of Home Assistant (Lovelace), it is not a limitation of any UFP camera. Once HA makes the stream, it is completely separate and handled by HA. It goes:

UFP -> RTSP/S -> HA -> ffmpeg (LL-HLS/HLS) -> client.

So, if you are streaming works correctly in Lovelace, then the UFP pipeline is working correctly. Something is going wrong between HA and your client. A lot of devices will require HTTP/2, valid HTTPS certs _and_ be publicly accessible to access, none of which does HA or this integration provides out of the box. You also want to make sure your TV supports HLS (or LL-HLS if you have the setting enabled).

**Reply by sjftech** [reply]
Okay, that's good to know, when I try to call the service, I get the below error:-

"Failed to call service camera.play_stream. camera.front_camera does not support play stream service"

Any ideas? Apologies, I'm new to HA.

**Reply by AngellusMortis** [reply]
Can you actually stream the cameras in Lovelace (and not get the choppy JPEG picture slideshow)? That sounds like it might not be configured correctly. 

https://www.home-assistant.io/integrations/stream
